hanjaelee/app.js

```diff
@@ -41,39 +41,191 @@ app.get('/ping', (req, res) => {
+  } catch (error) {
+    console.error('Failed to Signup', error)
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
+
+// 과제 3 게시글 등록하기
+app.post('/post', async (req, res) => {
+  try {
+    const { title, content, userId } = req.body
+
+    const rawQuery = `
+    INSERT INTO posts (
+      title,
+      content,
+      user_id
+    ) VALUES (?, ?, ?);`
+
+    const rawData = await database.query(rawQuery, [title, content, userId])
+
+    if (rawData['affectedRows']) {
+      return res.status(201).json({ message: 'postCreated' })
+    }
+
+    return res.status(202).json({ message: 'postnotcreated' })
+  } catch (error) {
+    console.error('Failed to Posting', error)
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
+
+// 과제 4 모든 게시물 가져오기
+app.get('/post', async (req, res) => {
+  try {
+    const rawQuery = `
+    SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      p.id AS postingId,
+      p.image_url AS postingImageUrl,
+      p.content AS postingContent
+    FROM posts AS p
+    INNER JOIN users AS u
+    ON p.user_id = u.id`
+
+    await database.query(rawQuery, (err, rows) => {
+      const result = {}
+      const data = rows
+      result['data'] = data
+      return res.status(200).json(result)
     })
+  } catch (error) {
+    console.error('Failed To Get Posts', error)
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
+
+// 과제 5 유저 한명이 작성한 모든 게시물 가져오기
+app.get('/post/:userId', async (req, res) => {
+  try {
+    const { userId } = req.params
+    const rawQuery = `
+    SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      JSON_ARRAYAGG(
+        JSON_OBJECT(
+          'postingId', p.id,
+          'postingImageUrl', p.image_url,
+          'postingContent', p.content)) AS posting
+    FROM posts AS p
+    INNER JOIN users AS u
+    ON p.user_id = u.id WHERE u.id = ?
+    GROUP BY u.id;`
+
+    const [result] = await database.query(
+      rawQuery, userId)
+    res.status(200).json({ data: result })
+  } catch (error) {
+    console.error('Failed get post for user', error)
+    return res.status(400).json({ message: error.sqlMessage })
+  }
+})
+
+// 과제 6 유저 한명이 작성한 게시글 내용 수정하기
+app.patch('/post', async (req, res) => {
+  try {
+    const { postId, userId, postContent } = req.body
+    let rawQuery = `
+      UPDATE posts
+      SET content = ?
+      WHERE id = ? AND user_id = ?;`
+    const rawData = await database.query(rawQuery, [postContent, postId, userId])
+
+    if (rawData['affectedRows']) {
+      rawQuery = `
+      SELECT
+        u.id AS userId,
+        u.name AS userName,
+        p.id AS postingId,
+        p.title AS postingTitle,
+        p.content AS postingContent FROM posts AS p
+      INNER JOIN users AS u
+      ON p.user_id = u.id
+      WHERE u.id = ? AND p.id = ?
+      LIMIT 1;`
+      const [result] = await database.query(rawQuery, [userId, postId])
```

Comment:
이 부분을 아래와 같이 수정 후 결과 확인 부탁드려요!
```suggestion
      const result = await database.query(rawQuery, [userId, postId])
```

Reply:
결과 정상적으로 나옵니다 배열로 결과 값을 넣으면 값이 하나만 들어가고  피드백 주신 코드로 하면 전부 들어가네요
원래 sql 문에서 의도한 결과값이 데이터가 이미 배열이라서 그런걸까요? 무슨 차이인지 설명 부탁드립니다.

Reply:
`const [result]`로 결과를 받게 되었을 때, 구조분해할당이 되어 쿼리문에서 하나의 값만 할당되게 됩니다! 그래서 모든 게시물의 값을 가져오기 위해서 배열로 가져오는 게 아닌, 변수로써 가져와야 하는 것입니다 ~